### ISSUE TITLE
DataList - aria labeling and aria-selected should be removed from li elements

### ISSUE THREAD

**Issue by thatblindgeye** [patternfly/patternfly-react#11391]
**Describe the enhancement or change**
- Per [naming role guidance from W3C](https://www.w3.org/WAI/ARIA/apg/practices/names-and-descriptions/#naming_role_guidance), listitems should not have aria labeling even if technically it may work. We should remove any aria-label, aria-labelledby, and/or aria-describedby from any `li` items in this component.
- The `aria-selected` attribute must be removed from the DataListItem component, as the attribute is not valid on `li` elements

**Is this request originating from a Red Hat product team? If so, which ones and is there any sort of deadline for this enhancement?**

**Any other information?**

**Comment by Mash707** [comment]
Hi @thatblindgeye I would like to work on this. I have already set up the project locally. Is this the file you are talking about?

![Image](https://github.com/user-attachments/assets/814eba29-b90f-4b16-9dea-b084a8c4ba61)

Do you have any other information that could help with this?
Also should I questions regarding the issue here or in slack?
Thank You.

**Comment by thatblindgeye** [comment]
@Mash707 yep that's the file! The updates should really just be removing some things from the `li` element within that file; we'd just want to make sure we're not removing anything being used for anything other than these attributes (e.g. `aria-labelledby` may be used beyond just being passed to the `li`). And either is fine! Whichever you'd prefer to ask any questions
